src/index/TextScoring.cpp

```diff
@@ -110,18 +95,17 @@ float ScoreData::getScore(WordIndex wordIndex, TextRecordIndex contextId) {
     docId = DocumentIndex::make(contextId.get());
   } else {
     auto it = docIdSet_.upper_bound(convertedContextId);
-    if (it == docIdSet_.end()) {
-      if (docIdSet_.empty()) {
-        AD_THROW("docIdSet is empty and shouldn't be");
-      }
-      LOG(DEBUG) << "Requesting a contextId that is bigger than the largest "
-                    "docId. contextId: "
-                 << contextId.get() << " Largest docId: " << *docIdSet_.rbegin()
-                 << std::endl;
-      return 0;
-    } else {
-      docId = *it;
-    }
+    AD_CORRECTNESS_CHECK(!docIdSet_.empty(),
+                         "docIdSet is empty and shouldn't be.");
+    AD_CORRECTNESS_CHECK(
+        !(it == docIdSet_.end()),
+        absl::StrCat("Requesting a contextId that is bigger than the largest "
+                     "docId. Requested contextId: ",
+                     contextId.get(),
+                     " Largest docId: ", docIdSet_.rbegin()->get(),
+                     " This hints on faulty input data for wordsfile.tsv and "
+                     "or docsfile.tsv"));
```

Comment:
The `StrCat` is performed even when the error is not thrown, which is a great waste of time. 
In this case you should pass a lambda as the argument of the Correctness_Check, which assembles the error message only if it is needed.